src/app/related-search/related-search.component.spec.ts

```diff
@@ -77,7 +77,18 @@ describe('RelatedSearchComponent', () => {
     fixture.detectChanges();
   });
 
-  it('should create', () => {
+  it('should create an instance', () => {
     expect(component).toBeTruthy();
   });
+
+  it('should display results if any, along with right keyword', () => {
+    let compiled = fixture.debugElement.nativeElement;
+    if (component.results.length > 0) {
```

Comment:
just try removing the conditional statement, you will find that tests fail. everytime. 
solution to this is:-
1) directly initialize resutls variable as shown in above image.
2) have a spy on search-service ( https://angular.io/guide/testing#test-a-component-with-an-async-service ) and get results by spying upon that